File: Leetcode/majority_element.py
# https://leetcode.com/problems/majority-element/
'''
    Author @Subhajit Das
    SWE Intern @Turbot HQ India PVT Ltd.
    University of Engineering and Management, Kolkata
    21/01/2020
'''


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        candidateKey = None

        for num in nums:
            if (count == 0):
                candidateKey = num

            count += 1 if num == candidateKey else -1
        return candidateKey

File: Leetcode/MajorityElement.java
// https://leetcode.com/problems/majority-element/
/*
    Author @Subhajit Das (sdsubhajitdas.github.io)
    SWE @Turbot HQ India PVT Ltd.
    15/05/2022
*/

import java.util.*;

class Solution {
  public int majorityElement(int[] nums) {
    int element=0,count=0;

    for(int num: nums) {
      if(count == 0) element = num;
      count += (element == num) ? 1 : -1;
    }
    return element;
  }
}